agent/command/archive_auto_extract_test.go

```diff
@@ -75,80 +84,100 @@ func (s *AutoExtractSuite) TestErrorWhenExcludeFilesExist() {
 	s.cmd.TargetDirectory = s.targetLocation
-	s.cmd.ArchivePath = filepath.Join(testutil.GetDirectoryOfFile(),
-		"interface.go")
+	_, thisFile, _, _ := runtime.Caller(0)
+	s.cmd.ArchivePath = thisFile
 
-	s.Error(s.cmd.Execute(context.Background(), s.comm, s.logger, s.conf))
+	s.Error(s.cmd.Execute(s.ctx, s.comm, s.logger, s.conf))
 }
 
-func (s *AutoExtractSuite) TestExtractionZipWorkingCase() {
-	s.conf.WorkDir = "/srv/evergreen"
+func (s *AutoExtractSuite) TestExtractionZipSucceedsButIsNotIdempotent() {
 	s.cmd.TargetDirectory = s.targetLocation
 	s.cmd.ArchivePath = filepath.Join(testutil.GetDirectoryOfFile(),
 		"testdata", "archive", "artifacts.zip")
 
-	s.NoError(s.cmd.Execute(context.Background(), s.comm, s.logger, s.conf))
+	s.NoError(s.cmd.Execute(s.ctx, s.comm, s.logger, s.conf))
 
-	counter := 0
-	err := filepath.Walk(s.targetLocation, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-		counter++
-		return nil
-	})
-	s.NoError(err)
-	s.True(counter > 1)
+	checkCommonExtractedArchiveContents(s.T(), s.cmd.TargetDirectory)
+
+	// Extracting the same archive contents multiple times to the same directory
+	// results in an error because the files already exist.
+	s.Error(s.cmd.Execute(s.ctx, s.comm, s.logger, s.conf))
 }
 
-func (s *AutoExtractSuite) TestExtractionTarWorkingCase() {
-	s.conf.WorkDir = "/srv/evergreen"
+func (s *AutoExtractSuite) TestExtractionTarSucceedsButIsNotIdempotent() {
 	s.cmd.TargetDirectory = s.targetLocation
 	s.cmd.ArchivePath = filepath.Join(testutil.GetDirectoryOfFile(),
 		"testdata", "archive", "artifacts.tar.gz")
 
-	s.NoError(s.cmd.Execute(context.Background(), s.comm, s.logger, s.conf))
+	s.NoError(s.cmd.Execute(s.ctx, s.comm, s.logger, s.conf))
 
-	counter := 0
-	err := filepath.Walk(s.targetLocation, func(path string, info os.FileInfo, err error) error {
+	checkCommonExtractedArchiveContents(s.T(), s.cmd.TargetDirectory)
+
+	// Extracting the same archive contents multiple times to the same directory
+	// results in an error because the files already exist.
+	s.Error(s.cmd.Execute(s.ctx, s.comm, s.logger, s.conf))
+}
+
+// checkCommonExtractedArchiveContents checks that the testdata's archive file
+// extracted to the expected contents.
+func checkCommonExtractedArchiveContents(t *testing.T, targetDirectory string) {
+	// There's a single file artifacts/dir1/dir2/testfile.txt.
+	expectedContents := map[string]bool{}
+	pathParts := []string{"artifacts", "dir1", "dir2", "testfile.txt"}
+	for i := range pathParts {
+		expectedContents[filepath.Join(pathParts[:i+1]...)] = false
+	}
+
+	// Check that the proper directory structure is created and contains the one
+	// extracted file.
+	err := filepath.Walk(targetDirectory, func(path string, info os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
-		counter++
+
+		assert.True(t, strings.HasPrefix(path, targetDirectory), "path '%s' should be inside target directory '%s'", path, targetDirectory)
```

Comment:
I intentionally made this an assertion. Ideally, the file traversal should reveal all the problem files rather than stop on the first one. I only use `require` if it's something that would prevent testing further conditions (e.g. requiring for nil pointers to avoid testing panics).